waku/protocol/v2/waku_protocol2.nim

```diff
@@ -53,11 +59,8 @@ method initPubSub*(w: WakuSub) =
 
   # Using GossipSub
   w.gossip_enabled = true
-
-  if w.gossip_enabled:
-    procCall GossipSub(w).initPubSub()
-  else:
-    procCall FloodSub(w).initPubSub()
+  
+  procCall w.pubsub().initPubSub()
```

Comment:
why do you want to disable dynamic dispatch here, you most certainly want the call graph to dispatch correctly here.

Reply:
Actually, this seems to be the issue. I believe (haven't tried), that by calling `procCall w.pubsub().initPubSub()` you are creating a loop, what you want is `procCall PubSub(w).pubsub().initPubSub()`, note the cast on the `w`, otherwise you are disabling dynamic dispatch, but keep calling the same method on the instance.

Reply:
Oh? You are probably right, however how come it worked before? Both flood and gossipsub behaviour was working.

Reply:
FWIW I believe I modelled this on existing usage in nim-libp2p/NBC, but maybe I did it wrong / generalized incorrectly

Reply:
It seems like what you want is `procCall GossipSub(w).pubsub().initPubSub()` here, since `WakuSub` inherits from `GossipSub` and `procCall` is essentially `.super()`.